### ISSUE TITLE
MTS-100 thermostatic valves: Specify auxiliary temperature sensor to override TRV temperature readings

### ISSUE THREAD

**Issue by ertechdesign** [krahabb/meross_lan#95]
**Is your feature request related to a problem? Please describe.**
The MTS-100 TRV built in temperature sensor is heated by the radiator itself and gives crazy readings which prevents me from controlling temperature in the room ie. it says the current temperature is 24 °C after 30 minutes of heating while the room was only heated to 20.5 °C.

**Describe the solution you'd like**
Ability to control the following settings of the TRV:
- cold_tolerance
- hot_tolerance
- target_sensor (that would replace the built-in thermometer in the TRV)

**Describe alternatives you've considered**
Is there any way to expose the YAML code for the entities created by your integration and add the following code?
`cold_tolerance: 0.8
hot_tolerance: 0
target_sensor: sensor.my_auxiliary_thermometer`

_EDIT:_ I'm able to specify "ambient_temperature" parameter on the frontend to display actual temperature in the climate entity but the TRV doesn't seem to use this reading to start/stop heating.

**Additional context**
Sorry if I'm dumb and missed something just there's no wiki to explain how to do that.

**Comment by krahabb** [comment]
Hello @ertechdesign ,
That would undoubtedly be useful and I'm going to think a bit about that. As per your suggestions/requests:

- use YAML code to customize components settings: no, meross_lan configuration is done all through the UI with a native HA feature called 'ConfigFlow' which is in place exactly to 'remove' the need for plain-text yaml configuration files. Nevertheless I could add support for what you're thinking about but I'll need to think a bit about it (going to do that)
- use a different 'ambient_temperature' to set the tracked temperature sensor on an external one: the TRV doesn't support tracking external values so that would mean the control code to be moved to HA. It would be exactly the same as creating a simple HA automation where you just turn on/off the valve in HA based on another sensor value. I would not go that way since if you really need this it means you can already do that this way (HA automations or maybe the native 'generic_thermostat') and likely have the knowledge to reliably implement it. Also, if this is the path, I guess advanced users like you could 'throw-in' also other conditions and behaviours in order to better customize the valve to their needs and so my basic implementation would soon fall behind.
- provide 'cold_tolerence' 'hot_tolerance': I'll see if the firmware valve itself supports this kind of corrections in order to leverage the device abilities else I guess my reasoning would 'fallback' to consider the 'generic_thermostat' platform again

I'll take some time to review and think about all of this

**Comment by krahabb** [comment]
I've tested a bit with the valve and found how to set the 'delta' for temperature compensation.
This way we could either configure a fixed offset, but that would likely be not ideal since the delta between the actual MTS reading and the room temperature is dependant on the state of the heating plant so when you're not heating the valve reading is more or less correct while it gets 'positively offset' when the radiator is heating and the amount of this offset would be changing with time and so.
I'm planning, as extrapolated from your suggestion, to offer both the ability to set a 'dumb' static offset but also, if you have another reliable sensor in place to 'bind' the valve reading to that sensor adjusting the offset constinuosly on the fly in order for the valve to always report and know (internally) what the real temperature is. This way, I expect the valve setpoint would follow the external sensor since it's actual value would be replicated internally into it's own temperature reading.
Let's see: it will take some time for me to implement since the 'hard part' is just building all of the UI flow in order to configure all of this stuff while the code would be relatively plain easy.
It shouldn't take very long tho. Maybe 1 week or so

**Comment by ertechdesign** [comment]
Wow, that's a really smart way to do that! I'm looking forward to it. 
I'm very happy to test this once released!
My setup:
1x Meross MSH-300 (Meross Hub)
4x Meross MTS-100 (TRVs)
4x Oria BLE thermo/hygrometer

**Comment by ertechdesign** [comment]
For the time being I was able to create a workaround directly in Home Assistant as @krahabb advised.

The main issue is `generic thermostat` only accepts a `switch` entity as a `heater`. My workaround was to create a switch that will control the Meross TRV.

Here's my code for `configuration.yaml`:

- Please populate `[SWITCH_NAME]` with any name you want
- Please populate `[MEROSS_TRV_ENTITY]` with the name of your existing Meross TRV `climate` entity name.   

```
switch:
  - platform: template
    switches:
      [SWITCH_NAME]:
        turn_on:
            service: climate.set_temperature
            data:
              temperature: 25
            target:
              entity_id: climate.[MEROSS_TRV_ENTITY]
        turn_off:
            service: climate.set_temperature
            data:
              temperature: 15
            target:
              entity_id: climate.[MEROSS_TRV_ENTITY]
```

- Please populate `[THERMOSTAT_NAME]` with the name of the new Thermostat
- Please populate `switch.[SWITCH_NAME]` with your selected switch name above
- Please populate `sensor.[AUXILIARY_THERMOMETER]` with any thermometer that you want to control your TRV with.

```
climate:
  - platform: generic_thermostat
    name: [THERMOSTAT_NAME]
    heater: switch.[SWITCH_NAME]
    target_sensor: sensor.[AUXILIARY_THERMOMETER]
    min_temp: 15
    max_temp: 25
    ac_mode: false
    cold_tolerance: 0.8
    hot_tolerance: 0
```

Now you can replace the Meross TRV with `climate.[THERMOSTAT_NAME]` in Home Assistant Front-End and control your TRV with [Generic Thermostat](https://www.home-assistant.io/integrations/generic_thermostat/). 

_Advantages:_
- The TRV stops giving false reading during heating
- You can use any thermometer you want that is supported in Home Assistant
- You can easily set minimum temperature, maximum temperature, hot and cold tolerance for advanced control.
- The TRV stops randomly adjusting the temperature 20 times a minute (this woke me up after a night out and annoyed me a lot) 

_Disadvantages:_
- I now have 8 thermostats for 4 TRVs (only 4 are exposed in the Home Assistent UI, the original TRVs are no longer required to be exposed)

_Principle of the solution:_
- When the generic thermostat starts heating, the TRV is set to 25 °C and gets fully opened
- When the generic thermostat stop heating, the TRV is set to 15 °C and gets fully closed

**Comment by krahabb** [comment]
the feature to set an external 'tracked sensor' has been implemented in latest releases (following #334)
